raft/log.go

```diff
@@ -791,6 +827,7 @@ loop:
 	// We don't add the +1 because the slice starts from 0.
 	quorumIndex := (nodeN / 2)
 	if quorumIndex >= len(indexes) {
+		l.tracef("sending heartbeat: no quorum: n=%d", quorumIndex)
```

Comment:
I must be missing something, but how can any of the code down here execute? There is an infinite loop above and I see no way to get past it.